LiteCore/Support/Timer.cc

```diff
@@ -37,18 +37,22 @@ namespace litecore::actor {
 
             } else if ( earliest->first <= clock::now() ) {
                 // A Timer is ready to fire, so remove it and call the callback:
-                auto timer        = earliest->second;
-                timer->_triggered = true;
-                _unschedule(timer);
+                _triggeredTimer             = earliest->second;
+                _triggeredTimer->_triggered = true;
+                _unschedule(_triggeredTimer);
 
                 // Fire the timer, while not holding the mutex (to avoid deadlocks if the
                 // timer callback calls the Timer API.)
                 lock.unlock();
                 try {
-                    timer->_callback();
+                    _triggeredTimer->_callback();
                 } catch ( ... ) {}
-                timer->_triggered = false;  // note: not holding any lock
-                if ( timer->_autoDelete ) delete timer;
+
+                if ( _triggeredTimer ) {
+                    _triggeredTimer->_triggered = false;  // note: not holding any lock
+                    if ( _triggeredTimer->_autoDelete ) delete _triggeredTimer;
```

Comment:
It appears to me that after the previous time, _triggeredTime may get deleted. Get the value of _triggeredTimer->_autoDelete before the previous line?

Reply:
Good point. I've pushed another commit to fix the race condition, and also to enforce that an autoDelete Timer is never manually deleted.